### ISSUE TITLE
Extend the extension mechanism

### ISSUE THREAD

**Issue by Ducasse** [pillar-markup/MicroDown#830]
In Microdown we can have extension in paragraph using the following syntax {!note|contents=should do something!}, {!citation|ref=Duca99a!}

It would be nice to be able to express them as {!citation|Duca99a!}

We should check if this is not already done. 
May be the extension argument logic could support this alternate way (where contents is implicit).

**Comment by Ducasse** [comment]
Task
- evaluate the current siuation, have a look at tests
- write some tests that show that we do have the support

**Comment by Ducasse** [comment]
This is related to #829 that should be done first

**Comment by Ducasse** [comment]
```
parse: delimiter stream: aTokenStream for: aParser

	| bodystring arguments tag extensionClass |
	bodystring := delimiter undelimitedSubstring.
	arguments := MicArgumentList
		             split: bodystring
		             defaultArg: #annotationKind
		             defaultValue: 'Me'.
	tag := arguments defaultValue trimBoth.

	extensionClass := self extensionClassFor: tag.
	self halt.
	optionalUsed := arguments 
		at: self optionalArgumentName 
		ifAbsent: [ true ] 
		ifPresent: [ false ].
	optionalUsed
		ifTrue: [ findOptional := arguments justTheArguments defaultValue key  ]
	
	^ extensionClass new
		  bodyString: bodystring;
		  arguments: arguments

```
